### ISSUE TITLE
Tooltip shows outside of plot area for scrollablePlotArea

### ISSUE THREAD

**Issue by wdrobinson** [highcharts/highcharts#13095]
#### Expected behaviour
Tooltip should not display outside of plot when using scrollablePlotArea = true or there should be a way to conditionally remove it. There are cases where having a legend below the chart is necessary and setting opacity to 1 works except the tooltip still shows when you mouse over the bottom of the chart. 

#### Live demo with steps to reproduce
https://jsfiddle.net/Lxo728a5/

#### Product version
Highcharts latest

![image](https://user-images.githubusercontent.com/10948156/76004765-0818a080-5ed8-11ea-8e0e-0ef326a27203.png)

**Comment by wdrobinson** [comment]
In addition to the bug of having the tooltip display where it shouldn't, if you scroll down halfway and move your mouse to the bottom of the chart below the legend so the tooltip causes the page to expand, the chart resizes which resets your scrollbar to the top.

**Comment by sebastianbochan** [comment]
Hi @wdrobinson,
Thank you for reporting about the problem which is caused by `tooltip.outside` parameter. Setting that as a false solves the problem.

Demo:
- https://jsfiddle.net/BlackLabel/Lkxehv8m/

Docs:
- https://api.highcharts.com/highcharts/tooltip.outside

**Comment by wdrobinson** [comment]
tooltip.outside is true by default for scrollable charts and in our case we need the outside tooltip for the chart with the issue so we need a fix or a better workaround for the bugs.

**Comment by wdrobinson** [comment]
Also even with changing tooltip.outside to false the tooltip appears at times because the .highcharts-scrollable-mask doesn't extend the entire width of the chart:
![image](https://user-images.githubusercontent.com/10948156/76352635-c3c93e00-62e5-11ea-9eb1-1c51c1cd7235.png)

https://jsfiddle.net/5j9tb4or/

**Comment by sebastianbochan** [comment]
You are right, now I see the problem. Thank you for the details!

**Comment by pawelfus** [comment]
*Internal note:*
There are some constraints in Tooltip (for scrollablePlotArea) already implemented. Check there what's missing to prevent rendering this tooltip for points outside of the visible area. Note that default kd-tree won't resolve that, because points are rendered within the given extremes.

**Comment by Izothep** [comment]
As a workaround, it should be possible to move function checking constraints outside of split tooltip, and then use it in in `Tooltip.refresh`:
http://jsfiddle.net/BlackLabel/n812y4mp/3/

___
Internal note: 

in default options we set boundaries as 
```
var bounds = {
    left: scrollLeft,
    right: scrollLeft + chartWidth,
    top: scrollTop,
    bottom: scrollTop + chartHeight
};
```
which is not limiting graph to plotArea, but to the whole graph. Is it intended? @pawelfus

**Comment by melloc01** [comment]
hi @TorsteinHonsi, there are legit cases for showing the Tooltip when the point is out of the chart box.

Take into consideration a line chart that has min and max set values and at one certain interval it goes out of these extremes, in this case the tooltip should show the value that is out of the box, or at least there should be an option for that. But I'd assume that showing should be the default behavior.


cc @lgdevelop

**Comment by KacperMadej** [comment]
Hi @melloc01 

tl;dr enabled in https://jsfiddle.net/BlackLabel/6w4mjh9a/1/

The current behavior is by design. The code ensures in multiple places that the tooltip for line (or line-like) series types won't show for points outside of the currently visible plot area. Some parts of the code could be overwritten and default options could be changed, but please note that the results are not fully tested and something might break either unexpectedly due to untested usage or after any code update (new release): https://jsfiddle.net/BlackLabel/6w4mjh9a/1/

If you would like to see a change in code you could open a new GitHub issue for an enhancement request and when the feature idea will gain enough popularity it will be implemented.

If you just looking for help with the problem you have described you could get help from technical support - https://www.highcharts.com/blog/support/

**Comment by melloc01** [comment]
hi @KacperMadej, at first I really thought the TL;DR was about my comment haha

Thanks for that, I had went to that code last week and that was one of my hidden escape hatches too, thanks for not letting me feel so bad about it now :) haha

Anyways I think this could be supported by the API, I just opened an issue for that, thanks!

https://github.com/highcharts/highcharts/issues/16662
